### PR TITLE
Kernel Mean Embeddings for all Half-Integer Matérn Kernels

### PR DESCRIPTION
**Pull request by marvinpfoertner**

### CODE REVIEW

File: src/linpde_gp/randprocs/covfuncs/_linfunctls.py [hunk 1]
```diff
@@ -118,15 +117,27 @@ def _(self, k: pn.randprocs.covfuncs.CovarianceFunction, /, *, argnum: int = 0):
...
+    if argnum not in (0, 1):
+        raise ValueError("`argnum` must either be 0 or 1.")
+
+    from ..crosscov.linfunctls.integrals import (  # pylint: disable=import-outside-toplevel
+        CovarianceFunction_Identity_LebesgueIntegral,
+    )
+
+    return CovarianceFunction_Identity_LebesgueIntegral(k, self, reverse=(argnum == 0))
+
+
```
Comment (timweiland) [inline src/linpde_gp/randprocs/covfuncs/_linfunctls.py:130]:
```suggestion

@LebesgueIntegral.__call__.register # pylint: disable=no-member
def _(self, k: covfuncs.StackCovarianceFunction, /, *, argnum: int = 0):
    from ..crosscov import (  # pylint: disable=import-outside-toplevel
        StackedProcessVectorCrossCovariance,
    )

    return StackedProcessVectorCrossCovariance(
        *(self(covfunc, argnum=argnum) for covfunc in k.covfuncs)
    )


```

LebesgueIntegral(CovarianceFunction) has precedence over LinearFunctional(StackedCovarianceFunction). This is also the reason for the error in the notebook. We can fix this by explicitly reintroducing the rule for `LebesgueIntegral`.

**Review by timweiland — APPROVED** [review]
Thanks! :pray: 

With the change I suggested, the CPU notebook still doesn't run through without errors. I think that's because Derivative(Integral) is still undefined. With Leibniz' rule, we can just pull it inside to get Integral(Derivative). I implemented this locally for partial derivatives, but it might make sense to do it more generally for any differential operator.

File: src/linpde_gp/randprocs/covfuncs/_linfunctls.py (hunk 1 above)
Comment (marvinpfoertner) [reply src/linpde_gp/randprocs/covfuncs/_linfunctls.py:135]: I think we can solve this problem a bit more generically. What do you think about the solution in #645c3a86?

Comment (timweiland) [reply src/linpde_gp/randprocs/covfuncs/_linfunctls.py:135]: Beautiful :+1:

**Pull request merged**